src/battle_interface.c

```diff
@@ -939,10 +939,65 @@ void DummyBattleInterfaceFunc(u8 healthboxSpriteId, bool8 isDoubleBattleBattlerO
 
 }
 
+static void TryToggleHealboxVisibility(u8 priority, u8 healthboxLeftSpriteId, u8 healthboxRightSpriteId, u8 healthbarSpriteId, u8 indicatorSpriteId)
+{
+    u8 spriteIds[4] = {healthboxLeftSpriteId, healthboxRightSpriteId, healthbarSpriteId, indicatorSpriteId};
+    int i;
+    
+    switch (gBattleResources->bufferA[gBattleAnimAttacker][0])
+    {
+    case CONTROLLER_MOVEANIMATION:
+        if (gBattleResources->bufferA[gBattleAnimAttacker][1] == MOVE_TRANSFORM)
+            return;
+        break;
+    case CONTROLLER_BALLTHROWANIM:
+        return;   //throwing ball does not hide hp boxes
+    case CONTROLLER_BATTLEANIMATION:
+        //check special anims that hide health boxes
+        switch (gBattleResources->bufferA[gBattleAnimAttacker][1])
+        {
+        case B_ANIM_TURN_TRAP:
+        case B_ANIM_LEECH_SEED_DRAIN:
+        case B_ANIM_MON_HIT:
+        case B_ANIM_SNATCH_MOVE:
+        case B_ANIM_FUTURE_SIGHT_HIT:
+        case B_ANIM_DOOM_DESIRE_HIT:
+        case B_ANIM_WISH_HEAL:
+        //new
+        case B_ANIM_MEGA_EVOLUTION:
+        case B_ANIM_TERRAIN_MISTY:
+        case B_ANIM_TERRAIN_GRASSY:
+        case B_ANIM_TERRAIN_ELECTRIC:
+        case B_ANIM_TERRAIN_PSYCHIC:
+            break;
+        }
+        return; //all other special anims dont hide
+    default:
```

Comment:
You maybe could instead do a fall-through to default here since default case is just a return as well? Just a suggestion rather than something required though.